test/analytics/__fixtures__/plugins/analytics_plugin_a/server/plugin.ts

```diff
@@ -79,17 +79,21 @@ export class AnalyticsPluginAPlugin implements Plugin {
     router.get(
       {
         path: '/internal/analytics_plugin_a/actions',
-        validate: {
-          query: schema.object({
-            takeNumberOfActions: schema.number({ min: 1 }),
-          }),
-        },
+        validate: false,
       },
       async (context, req, res) => {
-        const { takeNumberOfActions } = req.query;
-
-        const actions = await firstValueFrom(actions$.pipe(take(takeNumberOfActions), toArray()));
-
+        let found = false;
+        const actions = await firstValueFrom(
+          actions$.pipe(
+            takeWhile(() => !found),
+            tap(({ action, meta }) => {
+              found =
```

Comment:
Filter actions by the event type we're expecting, allowing us to be more specific about the actions (and their events) we want to assert against.

Reply:
See my previous comments for a potential use of `found$` :)